app/src/lib/git/for-each-ref.ts

```diff
@@ -74,10 +74,34 @@ export async function getBranches(
       continue
     }
 
+    const existsOnRemote =
+      upstream !== ''
+        ? await checkBranchExistsOnRemote(upstream, repository)
+        : false
+
     branches.push(
-      new Branch(name, upstream.length > 0 ? upstream : null, tip, type)
+      new Branch(
+        name,
+        upstream.length > 0 ? upstream : null,
+        tip,
+        type,
+        existsOnRemote
+      )
     )
   }
 
   return branches
 }
+
+async function checkBranchExistsOnRemote(
+  upstream: string,
+  repository: Repository
+): Promise<boolean> {
+  const result = await git(
+    ['show-ref', '--verify', 'refs/remotes/' + upstream],
```

Comment:
Could you provide some more context for this? My understanding of this is that we're checking to see if the remote exists. I'm not sure I understand what that has to do with checking if the remote _ref_ exists.

Reply:
It was my understanding that a branch is deleted on the remote the remote ref would be deleted after a fetch.

Although, see my comments here about using `ls-remote` instead https://github.com/desktop/desktop/pull/3600#discussion_r156976907

Reply:
Ah, I see. I misunderstood what was happening here 👍